common/src/main/java/com/wynntils/wynn/model/SpellModel.java

```diff
@@ -5,49 +5,75 @@
 package com.wynntils.wynn.model;
 
 import com.wynntils.core.WynntilsMod;
-import com.wynntils.core.components.Managers;
 import com.wynntils.core.components.Model;
 import com.wynntils.mc.event.SubtitleSetTextEvent;
+import com.wynntils.utils.StringUtils;
 import com.wynntils.wynn.event.SpellCastEvent;
+import com.wynntils.wynn.event.SpellEvent;
+import com.wynntils.wynn.event.SpellProgressEvent;
 import com.wynntils.wynn.model.actionbar.event.SpellSegmentUpdateEvent;
+import com.wynntils.wynn.objects.SpellDirection;
 import com.wynntils.wynn.objects.SpellType;
+import java.util.Arrays;
+import java.util.regex.MatchResult;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
 import net.minecraftforge.eventbus.api.SubscribeEvent;
 
 public class SpellModel extends Model {
-    private static final Pattern LEVEL_1_SPELL_PATTERN =
-            Pattern.compile("§a(Left|Right|\\?)§7-§a(Left|Right|\\?)§7-§r§a(Left|Right|\\?)§r");
-    private static final Pattern LOW_LEVEL_SPELL_PATTERN = Pattern.compile("§a([LR?])§7-§a([LR?])§7-§r§a([LR?])§r");
+    private static final Pattern SPELL_TITLE_PATTERN =
+            StringUtils.compileCCRegex("§([LR]|Right|Left)§-§([LR?]|Right|Left)§-§([LR?]|Right|Left)§");
```

Comment:
I'd actually prefer not to use compileCCRegex. In fact, I'd like to remove it. IntelliJ does not understand that it is a regexp pattern, in contrast to Pattern.compile, and will not assist you.

I also have a memory that I spent quite some effort trying to get down a correct regex for spell patterns. Hm, maybe that was for the action bar middle segment.